src/software/gui/full_system/widgets/ai_control.cpp

```diff
@@ -205,15 +206,7 @@ void setupChannelSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
                      static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
                      on_channel_changed);
 
-    auto on_parameter_value_changed = [spin_box](int new_value) {
-        // We block signals while setting the value of the spinbox so that we don't
-        // trigger the `on_spinbox_value_changed` function, which would set the
-        // parameter value again and deadlock on the parameter's internal mutex
-        spin_box->blockSignals(true);
-        spin_box->setValue(new_value);
-        spin_box->blockSignals(false);
-    };
-    channel_parameter->registerCallbackFunction(on_parameter_value_changed);
+    setup__SpinBox(spin_box, channel_parameter);
```

Comment:
```suggestion
    setupSpinBox(spin_box, channel_parameter);
```
according to the coding style guide, this should be camelCase

Reply:
The build is failing with the following error:
fatal error: software/gui/shared/parameters_spinbox.h: No such file or directory
Do you know why this could be the case? The file name/location is correct.

Reply:
yes, that means that the `deps` list in the `BUILD` file doesn't include a library that builds `software/gui/shared/parameters_spinbox.h`. Hint: look at `software/gui/shared/BUILD` and `software/gui/full_system/widgets/BUILD`

Reply:
Would be good if you did a `git fetch upstream` `git merge upstream/master` to bring your branch up to date, so you can get rid of the missing new lines

Reply:
Thanks for the help, I've fixed all the issues.